students_submissions/gcd_abanoubm1234567.py

```diff
@@ -0,0 +1,9 @@
+def gcd(a: int, b: int) -> int:
+    while b:
+        a, b = b, a % b
```

Comment:
I see. So you are using Euler's algorithm. Please explain in a comment how Euler's algorithm works for all the people who don't know.